### PR TITLE
API: min python version

### LINKED ISSUE #475: Deprecate Python 3.4, 3.5 and add 3.6 as minimum supported version
**Context**  
Python 3.4 has reached end-of-life, and Python 3.5 is security fixes only.

NumPy in its next release will drop support for Python 3.5. [ref](https://numpy.org/neps/nep-0029-deprecation_policy.html)

Moving to Python 3.6 as the minimum version enables use of f-strings which simplify formatting of strings and allows asyncio.

**Description**   
Deprecate or drop Python 3.4 and 3.5 support. Set 3.6 as minimum supported version.

### PR DESCRIPTION
**Pull request by tacaswell**
### Description of proposed changes    

Set minimum Python to be py36.  This means we can use fstrings :smile_cat: 

### Related issue(s)  

Fixes #475

### CODE REVIEW

**Comment by jameshadfield** [comment]
No compelling reason to use <3.6 from me, and fstring support will be nice.

cc @tsibley @huddlej are there any reasons we need 3.4 / 3.5 (pehaps related to docker builds?)

**Comment by tsibley** [comment]
:+1: for going to 3.6. The 3.4/3.5 minimum was just being extra conservative 18 months ago.

@jameshadfield The Docker base image already uses 3.6, so good there. (And it will move to 3.7 soon.)

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,18 @@
 from pathlib    import Path
 from setuptools import setup
+import sys
+
+min_version = (3, 6)
+
+if sys.version_info < min_version:
+    error = """
+Beginning with Matplotlib 3.1, Python {0} or above is required.
```
Comment (huddlej) [inline setup.py]: Should this say "Beginning with augur 6.5.0, ..."?

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+
+This may be due to an out of date pip.
+
+Make sure you have pip >= 9.0.1.
```
Comment (huddlej) [inline setup.py]: Is this message something specific to matplotlib or is it generally an issue users encounter with pip and changing between Python versions?

File: setup.py (hunk 1 above)
Comment (tacaswell) [reply setup.py]: Whoops sorry about that!

File: setup.py (hunk 2 above)
Comment (tacaswell) [reply setup.py]: This is a general problem with old versions of pip not knowing about the require_python meta data

**Comment by tacaswell** [comment]
@huddlej fixed!

**Pull request merged**